### PR TITLE
upgrade(hyperledger-indy): ambassador to ambassador edge stack

### PR DESCRIPTION
**Pull request by saikumarbommakanti**
This PR will provide the support of using latest ambassador edge-stack

changes:
• Updated all services to use the latest Ambassador Edge Stack routing rules.

fixes https://github.com/hyperledger/bevel/issues/2359

### CODE REVIEW

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 1]
```diff
@@ -39,12 +33,184 @@
...
+    - notest
+
+- name: Set Helm value for ambassador_eip
+  set_fact:
+    allocation_ips_stdout: "{{ allocation_ips.stdout | default('No value', true) }}"
+  when:
+    - allocation_ips is defined
+
+- name: Extract port numbers from the 'ports' string
+  set_fact:
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:110]: you are setting fact and registering it as well why ?

Comment (suvajit-sarkar) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:110]: where is the set fact extracted_ports being used ?

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 2]
```diff
@@ -39,12 +33,184 @@
...
+    - network.type == 'indy'
+
+- name: Create custom values for aes helm chart
+  vars:
+    ports: "{{ network.env.ambassadorPorts.ports | default([]) }}"
+    lbSourceRangeDefault:
+      - 0.0.0.0/0
+    loadBalancerSourceRanges: "{{ network.env.loadBalancerSourceRanges | default(lbSourceRangeDefault) }}"
+    elastic_ip: "{{ allocation_ips_stdout }}"
+    extracted_ports: "{{ ambassadorports_list }}"
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:125]: extracted_ports is defined here again.

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 3]
```diff
@@ -39,12 +33,184 @@
...
+  shell: |
+    from='{{ network.env.ambassadorPorts.portRange.from | default('') }}'
+    to='{{ network.env.ambassadorPorts.portRange.to | default('') }}'
+    if [ -z "$from" ] || [ -z "$to" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.portRange.from=${from} --set ambassador.otherPorts.portRange.to=${to}"
+    fi
+  register: ambassadorRange
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:144]: where are you using the variable ambassadorRange?

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 4]
```diff
@@ -39,12 +33,184 @@
...
+  shell: |
+    ports=$(echo '{{ network.env.ambassadorPorts.ports | default('') }}' | sed -e 's/\[/\{/' -e 's/\]/\}/')
+    terminalPorts='{{ terminal.stdout | default('') }}'
+    if [ -z "$ports" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.ports={'${ports},${terminalPorts}'}"
+    fi
+  register: ambassadorPortsIndy
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:159]: what is the use of ambassadorPortsIndy ?

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 5]
```diff
@@ -39,12 +33,184 @@
...
+    arr=( $(seq $from $to) )
+
+    for i in "${arr[@]}"
+    do
+      if [[ ! ${availablePortRange[*]} =~ ${i} ]]; then
+        echo "false" && exit 0
+      fi
+    done
+    echo "true"
+  register: indy_ports_available
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:183]: where is indy_ports_available used ?

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 6]
```diff
@@ -39,12 +33,184 @@
...
+  kubernetes.core.helm:
+    kubeconfig: "{{ kubeconfig_path }}"
+    name: "edge-stack"
+    chart_ref: datawire/edge-stack
+    chart_version: 8.7.2
+    release_namespace: "{{ proxy_namespace }}"
+    create_namespace: true
+    values:
+      namespace: "{{ proxy_namespace }}"
+      ambassador_Range: "{{ ambassadorRange.stdout }}"
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml:205]: ambassador_Range and extracted_ports what are these values for ? are they even defined in the aes chart?

File: platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl [hunk 7]
```diff
@@ -25,12 +31,14 @@ emissary-ingress:
     - name: https
       port: 443
       targetPort: 8443
-{% for port in ports or [] %}
+{% if extracted_ports is defined %}
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl:34]: will this not break other platforms passing ports ?

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-indy/charts/indy-node/templates/service.yaml [hunk 8]
```diff
@@ -30,4 +26,53 @@ spec:
...
+      from: SELF
+---
+apiVersion: getambassador.io/v3alpha1
+kind: TCPMapping
+metadata:
+  name: "{{ .Values.node.name }}-client-mapping"
+  namespace: {{ .Values.metadata.namespace }}
+spec:
+  port: {{ .Values.client.ambassadorPort }}
+  service:  "{{ .Values.client.name }}.{{ .Values.metadata.namespace }}:{{ .Values.service.ports.clientTargetPort }}"
```
[the remark this answers is not in the document]
Comment (saikumarbommakanti) [reply platforms/hyperledger-indy/charts/indy-node/templates/service.yaml]: fixed the service name

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/node/templates/node.tpl [hunk 9]
```diff
@@ -44,6 +44,7 @@ spec:
       port: {{ stewardItem.node.port }}
```
[the remark this answers is not in the document]
Comment (saikumarbommakanti) [reply platforms/hyperledger-indy/configuration/roles/create/helm_component/node/templates/node.tpl]: removed old ambassador annotations

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml [hunk 10]
```diff
@@ -39,12 +33,184 @@
...
     src={{ playbook_dir }}/../../../platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl
     dest={{ playbook_dir }}/../../../platforms/shared/configuration/build/aes-custom-values.yaml
+  when:
+    - network.type != 'indy'
+
+- name: Check if ambassador installed
+  k8s_info:
+    api_version: v1
+    kind: Service
+    name: ambassador
```
[the remark this answers is not in the document]
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: removed this checks

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml (hunk 1 above)
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: updated the setfacts to use in node tpl's

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml (hunk 2 above)
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: removed extracted_ports

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml (hunk 4 above)
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: removed ambassadorPortsIndy

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml (hunk 6 above)
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: updated files to update in the custom values file

File: platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml (hunk 5 above)
Comment (saikumarbommakanti) [reply platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml]: removed this variable

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/node/templates/node.tpl [hunk 11]
```diff
@@ -19,6 +19,8 @@ spec:
     metadata:
       name: {{ component_name }}
       namespace: {{ component_ns }}
+    proxy:
+      provider: ambassador
```
Comment (suvajit-sarkar) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/node/templates/node.tpl:23]: what if proxy is none ?

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]
Remove old ambassador helm chart

**Review by suvajit-sarkar — APPROVED** [review]

**Pull request merged**